src/privileged/trackers/framescript.js

```diff
@@ -70,12 +70,17 @@ addEventListener("DOMContentLoaded", function(e) {
       return urls.some(url => url.test(src));
     });
 
-    sendAsyncMessage("beforeunload", {telemetryData});
+    sendAsyncMessage("CookieRestrictions:beforeunload", {telemetryData});
   }, {once: true});
 
   content.window.addEventListener("unload", () => {
-    sendAsyncMessage("unload", {});
+    telemetryData.completeLocation = content.location.href;
+    if (docShell && docShell.document) {
+      telemetryData.num_blockable_trackers = docShell.document.numTrackersFound;
+    } else {
+      telemetryData.num_blockable_trackers = -1; // some sort of error here
```

Comment:
I think the solution to the -1 problem is just not to do this, since the framescript lifetime exceeds the document lifetime you need to absolutely expect the document not to be alive at this point, which means you would overwrite the `num_blockable_trackers` you collected in `onBeforeUnload`.

Reply:
Yeah, playing around with this a little more I'm pretty certain that this is the reason for the -1, so I would just reset that number after the unload happens (on on DOMContentLoaded) and not set it to `-1`